test/common/parseEslintConfigForGlobals.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const readline = require('readline');
+
+const searchLines = [
+  '  no-restricted-globals:',
+  '  node-core/prefer-primordials:',
+];
+
+const restrictedGlobalLine = /^\s{4}- name:\s?([^#\s]+)/;
+const closingLine = /^\s{0,3}[^#\s]/;
+
+module.exports = function parseEslintConfigForGlobals() {
+  const eslintConfig = readline.createInterface({
+    input: fs.createReadStream(path.join(__dirname, '..', '..', 'lib', '.eslintrc.yaml')),
+    crlfDelay: Infinity,
+  });
+
+  const globals = [process];
+
+  let isReadingGlobals = false;
+  eslintConfig.on('line', (line) => {
```

Comment:
The returned list will be incomplete, right? `globals` will be returned before this callback gets run.

Reply:
We don't need to read and parse synchronously for this to work, the returned array gets filled up line by line. The return list in incomplete until the file is parse entirely but that's not an issue at all. When the  `exit` event is triggered, the file would have been read completely.

Reply:
I think the tests are failing because of the race condition